docs/proposed/psa-driver-wrappers-codegen-migration-guide.md

```diff
@@ -0,0 +1,32 @@
+Migrating to an auto genrated psa_crypto_driver_wrappers.c file
+===============================================================
+
+**This is a specification of work in progress. The implementation is not yet merged into Mbed TLS.**
+
+This document describes how to migrate to the auto generated psa_crypto_driver_wrappers.c file.
+It is meant to give the library user migration guidelines while the Mbed TLS project tides over multiple minor revs of version 1.0, after which this will be merged into psa-driver-interface.md.
+
+## Introduction
+
+The design of the Driver Wrappers code generation is based on the design proposal https://github.com/ARMmbed/mbedtls/pull/5067
+During the process of implementation there might be minor variations wrt versioning and broader implementation specific ideas, but the design remains the same.
+
+## Prerequisites
+
+Python3 and Jinja2 rev 2.10.1
+
+## Feature Version
+
+1.0
+
+### What's critical for a migrating user
+
+The Driver Wrapper auto generation project is designed to use a python templating library ( Jinja2 ) to render templates based on drivers that are defined using a Driver descrioption JSON file(s).
+
+While that is the larger goal, for version 1.0 here's what's changed
+
+#### What's changed
+
+(1) psa_crypto_driver_wrappers.c will from this point on be auto generated.
+(2) The auto generation is based on the template file at scripts/data_files/driver_templates/psa_crypto_driver_wrappers.conf.
+(3) So while all driver wrapper templating support is yet to come in, the library user will need to patch into the template file as needed.
```

Comment:
I would rather expect that if a user wants features that the generator doesn't support yet, they would write their own. This would be a transitional option, which we'd discontinue once the generator is sufficiently flexible.

Reply:
If the user wants to change/ update something over the psa_crypto_driver_wrapper.c file they will need to patch the psa_crypto_driver_wrapper.conf template file. Changing all other makefiles to help generate this becomes unnecessary this way (I wouldn't want to make a segue for such presence). And there is no risk of the file being overwritten.

The intention of (3) here is that if a user wants to patch in some changes, that will need to be done over the template file. 
Once the feature is completely in lock, stock and barrel the template file will need very minimal inputs from us excepting cases where support for a new entry point is added. 

If this is not clear how would you like it reworded ? @gilles-peskine-arm

Reply:
If I have a working C file, can I just replace the template with it? Or do some characters/words/… need escaping?

Reply:
You could just replace the template file with the c file. 
The first rev 1.0 does the same, psa_crypto_driver_wrappers.conf is the same as psa_crypto_driver_wrappers.c.